### PR TITLE
Introduce parametrized tests

### PR DESCRIPTION
**Pull request by zhanpon**
Hi, thank you for sharing a great project!

This pull request proposes introducing parametrized tests. It improves readability and maintainability of unit tests by decoupling test data from test scenarios. Parametrized tests should be suitable for testing the implementation of constraints because the main concern is the correct mapping of inputs and outputs.

Caution: parametrized tests of JUnit 5 are currently an experimental feature ([official documentation](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests)). So [spock](http://spockframework.org/spock/docs/1.3/data_driven_testing.html) might be a better alternative.

If you agree to the proposal, I am happy to help to rewrite other tests.

### CODE REVIEW

**Comment by making** [comment]
@zhanpon Awesome. Supporting JUnit 5 is actually in my list and It's really helpful if you move it forward.

**Comment by zhanpon** [comment]
@making Great to hear! I will work on it.

**Comment by zhanpon** [comment]
@making I've done the refactoring of the tests in `am.ik.yavi.constraint`. Could you review the changes?

File: pom.xml [hunk 1]
```diff
@@ -65,12 +65,36 @@
             <version>3.0.2</version>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.4.2</version>
```
Comment (making) [inline pom.xml]: I'd use `<junit.version>`

File: pom.xml [hunk 2]
```diff
@@ -167,6 +191,14 @@
                     </excludes>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.22.0</version>
```
Comment (making) [inline pom.xml]: I'd use `<surefire-plugin.version>`

**Comment by making** [comment]
Using `@ValueSource` when possible looks more readable than `@MethodSource`.

**Comment by zhanpon** [comment]
Thanks for the comments. I rewrite them in two ways. Which do you prefer?

- `@CsvSource` (https://github.com/making/yavi/pull/19/commits/0325544f9d044e0402312726f39cdf3d8b24c583)

- `@ValueSource` (https://github.com/making/yavi/pull/19/commits/5e08597d17a55a38b5cbfcf5f9820f4ae54735ed)

**Comment by making** [comment]
I prefer type-safe @ValueSource

**Comment by zhanpon** [comment]
@making OK. If you like the commit https://github.com/making/yavi/commit/5e08597d17a55a38b5cbfcf5f9820f4ae54735ed, I will rewrite others in this style as well.

**Comment by making** [comment]
@zhanpon 👍

**Comment by zhanpon** [comment]
@making I've done the work. Could you review it again?

## Commit summary

Add JUnit 5 dependencies, and rewrite the tests in `am.ik.yavi.constraint` with JUnit Jupiter API.

- Introduce `@ParametrizedTest` for all test class except `BooleanConstraintTest`, `ObjectConstraintTest`, `CollectionConstraintTest`, and `MapConstraintTest`.

- Extract `retrievePredicate` helper method to remove duplicate set-up code.

**Comment by making** [comment]
LGTM. Thanks!

**Pull request merged**